tfplan/json_plan_test.go

```diff
@@ -0,0 +1,51 @@
+package tfplan
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadJSONResources(t *testing.T) {
+	data := []byte(`
+{
+  "planned_values": {
+    "root_module": {
+      "child_modules": [
```

Comment:
Could we add a deeply nested test (2+ levels) just to confirm we don't introduce subtle bugs with recursion?

Reply:
done.

Reply:
actually the test is on 2 level already (level 1 has no resource, level 2 has one), now I added an extra level so as to make it more resilient.